src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1438,6 +1443,33 @@ public static bool IsInstanceContext(this SyntaxTree syntaxTree, int position, S
             return false;
         }
 
+        private static ArrowExpressionClauseSyntax GetExpressionBodyFromMember(MemberDeclarationSyntax containingMember)
+        {
+            if (containingMember != null)
+            {
+                switch (containingMember.Kind())
+                {
+                    case SyntaxKind.PropertyDeclaration:
+                        return ((PropertyDeclarationSyntax)containingMember).ExpressionBody;
+                    case SyntaxKind.MethodDeclaration:
+                        return ((MethodDeclarationSyntax)containingMember).ExpressionBody;
+                    case SyntaxKind.IndexerDeclaration:
+                        return ((IndexerDeclarationSyntax)containingMember).ExpressionBody;
+                    default:
```

Comment:
Expression-Bodies are supported on both Operators and ConversionOperators as well but Operator and Conversion Operators are both static. 'this' keyword must not get listed here. 

If this is made a helper, this will need Constructor and Destructor Syntaxes as well for being MemberDeclarationSyntaxes. At that point we will have code which will never get run.

May be I can rename the method name from `GetExpressionBodyFromMember` to `GetExpressionBodyFromMemberForThisKeywordRecommendation`